lib/ansible/parsing/dataloader.py

```diff
@@ -233,35 +236,33 @@ def path_dwim(self, given):
         given = unquote(given)
         given = to_text(given, errors='surrogate_or_strict')
 
-        if given.startswith(u"/"):
-            return os.path.abspath(given)
-        elif given.startswith(u"~"):
-            return os.path.abspath(os.path.expanduser(given))
+        if given.matches(os.path.sep) or given.matches(u'~'):
+            path = unfrackpath(given, follow=False)
         else:
             basedir = to_text(self._basedir, errors='surrogate_or_strict')
-            return os.path.abspath(os.path.join(basedir, given))
+            path = unfrackpath(os.path.join(basedir, given), follow=False)
+
```

Comment:
The clearer that I was talking about earlier was that the unfrackpath is called in both branches of the conditional.  So it's clearer if that happens after exiting the conditional.